app/Observers/HR/SlotsObserver.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Observers\HR;
+
+use App\Models\HR\Slots;
+use Illuminate\Http\Request;
+use Carbon\Carbon;
+
+class SlotsObserver
+{
+    /**
+     * Listen to the ApplicationMeta create event
+     * @param  ApplicationMeta $applicationMeta
+     * @return void
+     */
+    protected $request;
+
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+    }
+    public function created(Slots $slot)
+    {
+        if (!$slot->slot()->exists()) {
+            $request=$this->request;
+            $start_time=Carbon::parse($slot->starts_at);
+            $end_time=Carbon::parse($slot->ends_at);
+            $repeat_till=Carbon::parse(request('repeat_till'));
+            
+            switch (request('recurrence')) {
+                case 'weekly':
+                    $no_of_slots=$start_time->diffInWeeks($repeat_till);
```

Comment:
```suggestion
                    $noOfSlots=$start_time->diffInWeeks($repeat_till);
```

Please update all other variables